hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java

```diff
@@ -124,11 +124,11 @@ private byte[] encodeValue(VALUE value) throws IOException {
   }
 
   private KEY decodeKey(byte[] key) throws CodecException {
-    return key == null ? null : keyCodec.fromPersistedFormat(key);
+    return key != null && key.length > 0 ? keyCodec.fromPersistedFormat(key) : null;
   }
 
   private VALUE decodeValue(byte[] value) throws CodecException {
-    return value == null ? null : valueCodec.fromPersistedFormat(value);
+    return value != null && value.length > 0 ? valueCodec.fromPersistedFormat(value) : null;
```

Comment:
Hi @szetszwo @adoroszlai, is the `value.length > 0` check necessary here?
It seems to cause an issue in [HDDS-13302](https://issues.apache.org/jira/browse/HDDS-13302).
FSORepairTool writes zero-length byte arrays into the table when there are reachable files/dirs.
With this check, the method returns null, making it indistinguishable from a missing key. This means we can no longer tell whether the key exists with a zero-length byte arrays or doesn't exist at all.

Reply:
@chungen0126 , that's a good point!  We should properly support empty array/String.  Filed HDDS-13317.